File: solutions/c/isogram/1/isogram.c
#include "isogram.h"
#include <ctype.h>
#include <stddef.h>
#include <stdlib.h>
#include <string.h>

bool is_isogram(const char phrase[]) {
  if (!phrase)
    return false;

  size_t len = strlen(phrase);
  int *count = calloc(26, sizeof(int));

  for (size_t i = 0; i < len; i++) {
    char c = phrase[i];
    if (isalpha(c)) {
      c = tolower(c);
      int index = c - 97;
      count[index]++;

      if (count[index] > 1) {
        return false;
      }
    }
  }
  return true;
}

File: solutions/go/isogram/1/isogram.go
package isogram

import "strings"

func IsIsogram(word string) bool {
	letters := 0
	for _, ch := range strings.ToLower(word) {
		if ch < 'a' || ch > 'z' {
			continue
		}
		id := 1 << int(ch-'a')
		if letters&id != 0 {
			return false
		}
		letters |= id
	}
	return true
}
